monai/transforms/io/array.py

```diff
@@ -514,3 +515,46 @@ def __call__(
             "    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies.\n"
             f"   The current registered writers for {self.output_ext}: {self.writers}.\n{msg}"
         )
+
+
+class MappingJson(Transform):
+    """
+    Writes a JSON file that logs the mapping between input image paths and their corresponding output paths.
+
+    Args:
+        mapping_json_path (Path or str): Path to the JSON file where the mappings will be saved.
+    """
+
+    def __init__(self, mapping_json_path: Path | str = "mapping.json"):
+        self.mapping_json_path = Path(mapping_json_path)
+
+    def write_json(self, input_path: str, output_path: str):
+        """
+        Args:
+            input_path (str): The path of the input image file.
+            output_path (str): The path of the output image file.
+        """
+        log_data = {"input": input_path, "output": output_path}
+        try:
+            with self.mapping_json_path.open("r") as f:
+                existing_log_data = json.load(f)
+        except FileNotFoundError:
+            existing_log_data = []
+
+        existing_log_data.append(log_data)
+
+        with self.mapping_json_path.open("w") as f:
+            json.dump(existing_log_data, f, indent=4)
+
+    def __call__(self, img: MetaTensor):
+        """
+        Args:
+            img (MetaTensor): The input image with metadata.
+        """
+        if "saved_to" not in img.meta:
+            raise KeyError("Missing 'saved_to' key in metadata. Check SaveImage savepath_in_metadict.")
+
+        input_path = img.meta["filename_or_obj"]
```

Comment:
For the string `filename_or_obj`, I would recommand to use the enum [ImageMetaKey.FILENAME_OR_OBJ](https://github.com/Project-MONAI/MONAI/blob/2e53df78e580131046dc8db7f7638063db1f5045/monai/utils/misc.py#L577). For the `saved_to` string, @KumoLiu  I would suggest to create a new enum to record metadata keys that are optional. What do you think about this?

Thanks  
Bin

Reply:
If @KumoLiu thinks this is feasible, I will modify `MappingJson` and use ImageMetaKey.

Reply:
Agreed, and for `saved_to`, we can add it in https://github.com/Project-MONAI/MONAI/blob/e85580af2267404ff0f022e5372a44e8effe6316/monai/utils/enums.py#L536